python/lsst/ts/externalscripts/auxtel/latiss_intra_extra_focal_data.py

```diff
@@ -0,0 +1,172 @@
+                type: number
+                default: 0.0
+              offset_y:
+                description: y degree of freedom offset in mm.
+                type: number
+                default: 0.0
+              offset_z:
+                description: z degree of freedom offset in mm.
+                type: number
+                default: 0.0
+              offset_Rx:
+                description: Rx degree of freedom offset in degrees.
+                type: number
+                default: 0.0
+              offset_Ry:
+                description: Ry degree of freedom offset in degrees.
+                type: number
+                default: 0.0
+              offset_m1:
+                description: M1 pressure degree of freedom offset in Pa.
+                type: number
+                default: 0.0
+                maximum: 0.0
+        additionalProperties: false
+        """
+        schema_dict = yaml.safe_load(schema)
+        base_schema_dict = super(LatissIntraExtraFocalData, cls).get_schema()
+
+        for prop in base_schema_dict["properties"]:
+            schema_dict["properties"][prop] = base_schema_dict["properties"][prop]
+
+        return schema_dict
+
+    async def additional_configuration(self, config: types.SimpleNamespace) -> None:
+        """Additional configuration.
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        # set dofs
+        self.offset_x = config.offset_x
+        self.offset_y = config.offset_y
+        self.offset_z = config.offset_z
+        self.offset_Rx = config.offset_Rx
+        self.offset_Ry = config.offset_Ry
+        self.offset_m1 = config.offset_m1
+
+    async def arun(self, checkpoint: bool = False):
+        """Takes Intr and Extra Focal Data.
+
+        Parameters
+        ----------
+        checkpoint : `bool`, optional
+            Should issue checkpoints (default=False)?
+
+        """
+
+        await self._configure_target()
+
+        await self._slew_to_target(checkpoint)
+
+        if self.take_detection_image:
+            if checkpoint:
+                await self.checkpoint("Detection image.")
+            await self.latiss.take_acq(
+                self.acq_exposure_time,
+                group_id=self.group_id,
+                reason="DETECTION_INFOCUS"
+                + ("" if self.reason is None else f"_{self.reason}"),
+                program=self.program,
+            )
+
+        # Apply degrees of freedom
+        await self.hexapod_offset(
+            self.offset_z,
+            self.offset_x,
+            self.offset_y,
+            self.offset_Rx,
+            self.offset_Ry,
+            self.offset_m1,
+        )
+
+        # Setting visit_id's to none so run_cwfs will take a new dataset.
+        self.intra_visit_id = None
+        self.extra_visit_id = None
+        await self.take_intra_extra()
```

Comment:
Just to be in the safe side, maybe add a "try/except" clause here to make sure resetting the offset always happens:

```
try:
    await self.take_intra_extra()
finally:
    await self.hexapod_offset(...)
```